### DISCUSSION TITLE
SetUserAgentOverrideAsync Iframe not work

### DISCUSSION THREAD

**Discussion by drRobertDev** [cefsharp/CefSharp#4849] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

126.2.70

### Operating System

Windows 11

### Architecture

x64

### .Net Version

.net 8

### Implementation

OffScreen

### Reproduction Steps

UserAgentMetadata AgentMetaData =>
        new()
        {
            Brands = new List<UserAgentBrandVersion>()
            {                
                new UserAgentBrandVersion { Brand = "Not/A)Brand", Version = "8" }, 
                new UserAgentBrandVersion { Brand = "Chromium", Version = "126" },                                                                                                 
                new UserAgentBrandVersion { Brand = "Google Chrome", Version = "126" },
            },
            FullVersion = "126.0.0.0",
            Platform = "Windows",
            PlatformVersion = "Win64",
            Architecture = "x64",
            Model = "",
            Mobile = false,
        };

using var devToolsClient = _webController.GetDevToolsClient();
await devToolsClient.Network.SetUserAgentOverrideAsync(BrowserAgentChrome, userAgentMetadata: WebHelper.AgentMetaData);
await devToolsClient.Emulation.SetUserAgentOverrideAsync(BrowserAgentChrome, userAgentMetadata: WebHelper.AgentMetaData);

            _webController.LoadHtml("<!DOCTYPE html><html lang=\"en\"><head><meta charset=\"UTF-8\"><meta name=\"viewport\"content=\"width=device-width, initial-scale=1.0\"><title>Iframe Example</title></head><body><iframe src=\"https://easports-api.arkoselabs.com/v2/2.6.1/enforcement.bf6c920f19fb49ce16c82aaef4b83f45.html#A4EECF77-AC87-8C8D-5754-BF882F72063B&6777600a-9cef-4e3a-bde8-935a90ea7c27\"title=\"Ark\"></iframe></body></html>");

Look like not work with **Iframe** on page, then will show cef base value **sec-ch-ua: "Not/A)Brand";v="8", "Chromium";v="126"** instead **sec-ch-ua: "Not/A)Brand";v="8", "Chromium";v="126", "Google Chrome";v="126"** on iframe calls

.html load with exspected header, .js load with wrong header.

i tried also with             _webController.FrameLoadEnd += Browser_FrameLoadEnd;

private static async void Browser_FrameLoadEnd(object sender, FrameLoadEndEventArgs e)
{
    var browser = (ChromiumWebBrowser)sender;
    var frameDevToolsClient = e.Browser.GetDevToolsClient();

   await frameDevToolsClient.Network.SetUserAgentOverrideAsync(BrowserAgentChrome, userAgentMetadata: AgentMetaData);
    await frameDevToolsClient.Emulation.SetUserAgentOverrideAsync(BrowserAgentChrome, userAgentMetadata: AgentMetaData);
}

but look like not apply on iframe

### Expected behavior

sec-ch-ua: "Not/A)Brand";v="8", "Chromium";v="126", "Google Chrome";v="126"

### Actual behavior

sec-ch-ua: "Not/A)Brand";v="8", "Chromium";v="126"

### Regression?

not work in previous versions

### Known Workarounds

nope

### Does this problem also occur in the CEF Sample Application

Yes using WPF/OffScreen command line args

**Comment by amaitland** [comment]
Converting to discussion as cefsharp is just a wrapper. 

You'd have to work out if the problem is in chromium or in CEF and raise an issue in the relevant issue tracker.

 Read the first few paragraphs of https://github.com/cefsharp/CefSharp/wiki/General-Usage for background.

Have you considered puppeteer sharp?
